### PR TITLE
Fix time zone rule example

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by sffc — APPROVED** [review]

**Review by sffc — CHANGES_REQUESTED** [review]
Issue: This PR contains a new submodule at utils/diplomat; please do that in its own PR

**Comment by robertbastian** [comment]
Ugh it's keeps popping up in other branches

File: utils/resb/src/binary.rs [hunk 1]
```diff
@@ -69,6 +69,7 @@ struct BinReprInfo {
 /// The `BinIndex` struct represents details of the written bundle.
 ///
 /// The index is present from [`FormatVersion::V1_1`] on.
+#[allow(dead_code)]
```
Comment (sffc) [inline utils/resb/src/binary.rs:72]: @leftmostcat any idea why this is dead code?

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
```
error: the borrowed expression implements the required traits
   --> utils/resb/examples/time_zone_rule.rs:150:34
    |
150 |       let in_bytes = std::fs::read(&Path::new(concat!(
    |  __________________________________^
151 | |         env!("CARGO_MANIFEST_DIR"),
152 | |         "/examples/data/zoneinfo64.res"
153 | |     )))
    | |______^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
    = note: `-D clippy::needless-borrow` implied by `-D warnings`
help: change this to
    |
150 ~     let in_bytes = std::fs::read(Path::new(concat!(
151 +         env!("CARGO_MANIFEST_DIR"),
152 +         "/examples/data/zoneinfo64.res"
153 ~     )))
    |
```

File: utils/resb/src/binary.rs (hunk 1 above)
Comment (leftmostcat) [reply utils/resb/src/binary.rs:72]: Several of these fields are only read from at present with the `serialize` feature enabled. Looks like if I run `cargo check`, I get a warning about that which I don't get with `cargo check --all-features`.

Comment (sffc) [reply utils/resb/src/binary.rs:72]: Ok, can you fix it in your follow-up PR by gating this on `#[cfg(feature = "serialize")]` or whichever feature needs it?

**Review by sffc — APPROVED** [review]

**Pull request merged**